src/GoTrueClient.ts

```diff
@@ -580,33 +580,43 @@ export default class GoTrueClient {
   > {
     const storageItem = await getItemAsync(this.storage, `${this.storageKey}-code-verifier`)
     const [codeVerifier, redirectType] = ((storageItem ?? '') as string).split('/')
-    const { data, error } = await _request(
-      this.fetch,
-      'POST',
-      `${this.url}/token?grant_type=pkce`,
-      {
-        headers: this.headers,
-        body: {
-          auth_code: authCode,
-          code_verifier: codeVerifier,
-        },
-        xform: _sessionResponse,
+
+    try {
+      const { data, error } = await _request(
+        this.fetch,
+        'POST',
+        `${this.url}/token?grant_type=pkce`,
+        {
+          headers: this.headers,
+          body: {
+            auth_code: authCode,
+            code_verifier: codeVerifier,
+          },
+          xform: _sessionResponse,
+        }
+      )
+      await removeItemAsync(this.storage, `${this.storageKey}-code-verifier`)
+      if (error) {
+        throw error
```

Comment:
Should we continue to return the error here?  Seems like we return the error in the [original version](https://github.com/supabase/auth-js/pull/946/files#diff-3522461172efd6058d6b8da62fc2d30d8b524d2b64894ea2c67218c52f7fdff5L598) so wanted to confirm it is intended

`{ data: { user: null, session: null, redirectType: null }, error }`

Let me know if there are considerations I've missed though

Reply:
@J0 it will get caught in the catch block below and returned

Reply:
Perfect, thanks for the clarification